File: ios/HeightMeasure/Models/MeasurementResult.swift
import Foundation
import simd

struct MeasurementResult: Identifiable, Codable, Equatable {
    let id: UUID
    let verticalHeight: Float
    let euclideanDistance: Float
    let category: HeightCategory
    let bottomPoint: SIMD3<Float>
    let topPoint: SIMD3<Float>
    let timestamp: Date

    var showsSlantDistance: Bool {
        abs(euclideanDistance - verticalHeight) > 0.02
    }
}

File: unity/HeightMeasure/Assets/Scripts/Measurement/MeasurementResult.cs
using UnityEngine;
using System;

[Serializable]
public class MeasurementResult
{
    public float verticalHeight;
    public float euclideanDistance;
    public HeightCategory category;
    public Vector3 bottomPoint;
    public Vector3 topPoint;
    public DateTime timestamp;

    public bool ShowsSlantDistance => Mathf.Abs(euclideanDistance - verticalHeight) > 0.02f;
}
